packages/argon2-cffi/meta.yaml

```diff
@@ -0,0 +1,22 @@
+package:
+  name: argon2-cffi
+  version: 23.1.0
+  top-level:
+    - argon2
+source:
+  url: https://files.pythonhosted.org/packages/a4/6a/e8a041599e78b6b3752da48000b14c8d1e8a04ded09c88c714ba047f34f5/argon2_cffi-23.1.0-py3-none-any.whl
+  sha256: c670642b78ba29641818ab2e68bd4e6a78ba53b7eff7b4c3815ae16abf91c7ea
```

Comment:
Instead of including the wheel here, could we include the sdist and add your PR https://github.com/hynek/argon2-cffi/pull/190 as a patch with a short comment about it? Since `argon2-cffi` is pure Python, it shouldn't take more than ~2-3 seconds to build from the sdist, and we—or you, if you get to it first—will remember to drop the patch as soon as it gets merged and released.

Reply:
I moved to the sdist. I also added i run depenency for the binding one.
 
I also disabled a test because i couldn't (?) access the _pytest_ package to use the `pytest.raises` function. Any idea?

Reply:
Ah, you may add the `import pytest` statement back there, and it should all work fine; I wonder if we need to add something about it to the docs, or whether this was something you faced locally?

Reply:
I added back the test. For the docs, I added a message to the error providing context (i propagated the change also in the upstream PR). Now the user will have something like
```py
>>> import argon2
>>> ph = argon2.PasswordHasher(parallelism=2)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/lib/python3.12/site-packages/argon2/_password_hasher.py", line 138, in __init__
    raise UnsupportedParamsError(msg)
argon2.exceptions.UnsupportedParamsError: within wasm/wasi environments `parallelism` must be set to 1
>>>
```
Maybe we should add something in the FAQ about the global problem something like

```
Q: Can i have multi-threading with pthread?
A: No, at the moment all libraries using plain multithreading will have to use 1 thread or add specific support for the platform
```

Reply:
Thanks, the error message makes sense to me!

> Maybe we should add something in the FAQ about the global problem something like
> 
> ```
> Q: Can i have multi-threading with pthread?
> A: No, at the moment all libraries using plain multithreading will have to use 1 thread or add specific support for the platform
> ```

Yes, and we should add a small admonition to this section as well: https://pyodide.org/en/stable/development/new-packages.html#determining-if-creating-a-pyodide-package-is-necessary so that people check if the project they are trying to package will be eventually possible to support (in case it makes heavy use of `multiprocessing` or `threading`).

Reply:
Thanks, I updated the documentation in the last commit. 
Marking this conversation as resolved.